### PR TITLE
Bar charts and paneling

### PR DESCRIPTION
**Pull request by samussiah**

### CODE REVIEW

File: src/chart/summaryTable/renderRow/makeBarChart.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                ,'alignment-baseline': 'middle'
+                ,'dx': '1em'
+                ,'x': this.plot_width})
+            .text(d => d.length < 30
+                ? d
+                : d.substring(0,30)+'...');
+
+      //Draw overall rates.
+        if (this.config.group) {
+            this.current_data
```
Comment (jwildfire) [inline src/chart/summaryTable/renderRow/makeBarChart.js]: So this extra mark is showing the overall rate for the given level *within the context of the selected filters*, right? 

For example, if we are looking at the chart for race (bars are White, Black, etc) grouped by gender (panels for male and female), filtered to a single site (Site 1). The marks for white participants in the females panel would be showing the % of the female population in site 1 that is white (the bar) vs. the total percent of the population (both male and female) in site 1 that is white (the line). Correct?

In other words, the line the example above is the same as the bar in the default example with no group (assuming the same filters)? 

If I'm understanding everything right, I'd suggest keeping that 'overall' bar visually identical for the ungrouped and grouped example. maybe {stroke:black, fill: white, height: x} for the 'overall' mark. and then {fill:black (or some color), height: 0.8*x} for the grouped data]

File: src/chart/summaryTable/renderRow/makeBarChart.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+                            .style(
+                                {'stroke': 'black'
+                                ,'stroke-width': '3px'
+                                ,'stroke-opacity': '1'});
+                        rateLine
+                            .append('title')
+                            .text(`Overall rate: ${d3.format('.1%')(x)}`);
+
+                      //Draw line from group rate to overall rate of the y-axis value.
+                        const diffLine = g
```
Comment (jwildfire) [inline src/chart/summaryTable/renderRow/makeBarChart.js]: I think I'd just make this visible when one of the relevant bars is hit with a mouseover.

**Comment by jwildfire** [comment]
Also - might be worth throwing in a y-axis sort to keep the order the same (see 2nd panel below) screen shot 2017-04-07 at 11 47 57 am

File: src/chart/summaryTable/renderRow/makeBarChart.js (hunk 1 above)
Comment (samussiah) [inline src/chart/summaryTable/renderRow/makeBarChart.js]: Yeah, I think you described it right, but just to make sure we're on the same page: the position of the overall mark in grouped view for a given categorical value should match the position of the right side of the that categorical value's bar in the ungrouped view.
`
Ungrouped:
-----| F (60%)
---|   M (40%)
Grouped:
Site 1 (n=10)
-----|  F (50%)
---|-   M (50%)
Site 2 (n=10)
-----|- F (70%)
---|    M (30%)
`

Comment (jwildfire) [inline src/chart/summaryTable/renderRow/makeBarChart.js]: yep, I think we're on the same page. And overall - I think this is great!

The proposed change is to make the line in the grouped view in to a bar that is visually consistent with the overall (ungrouped) view.

File: src/chart/summaryTable/renderRow/makeBarChart.js (hunk 2 above)
Comment (samussiah) [inline src/chart/summaryTable/renderRow/makeBarChart.js]: Sounds good to me.

**Comment by samussiah** [comment]
Agreed, the subgroup y-domains should match the overall y-domain.

**Comment by samussiah** [comment]
The x-domain though...  it should encompass both the greatest group rate and the overall group rate.

**Comment by jwildfire** [comment]
Yes - both x and y domains should be the same for all panels

**Comment by samussiah** [comment]
Fixes in.

**Comment by samussiah** [comment]
@jwildfire this branch has fully functional bar charts and dot plots, and the histogram code has been relocated from `src` to `src\chart\summaryTable\renderRow\makeChart`.
